src/stdlib/SDL_malloc.c

```diff
@@ -2531,9 +2531,9 @@ static int init_mparams(void) {
       int fd;
       unsigned char buf[sizeof(size_t)];
       /* Try to use /dev/urandom, else fall back on using time */
-      if ((fd = open("/dev/urandom", O_RDONLY)) >= 0 &&
-          read(fd, buf, sizeof(buf)) == sizeof(buf)) {
-        s = *((size_t *) buf);
+      if (fd = open("/dev/urandom", O_RDONLY) >= 0) {
+        if (read(fd, buf, sizeof(buf)) == sizeof(buf))
```

Comment:
since you're being overly cautious: you aren't handling the case where `read()` might fail, i.e. you aren't jumping to the else as the original code did.

Reply:
Maybe the following patch instead?
```diff
diff --git a/src/stdlib/SDL_malloc.c b/src/stdlib/SDL_malloc.c
index 8238e8b..fa77731 100644
--- a/src/stdlib/SDL_malloc.c
+++ b/src/stdlib/SDL_malloc.c
@@ -2531,11 +2531,14 @@ static int init_mparams(void) {
       int fd;
       unsigned char buf[sizeof(size_t)];
       /* Try to use /dev/urandom, else fall back on using time */
-      if ((fd = open("/dev/urandom", O_RDONLY)) >= 0 &&
-          read(fd, buf, sizeof(buf)) == sizeof(buf)) {
-        s = *((size_t *) buf);
+      if ((fd = open("/dev/urandom", O_RDONLY)) < 0)
+        s = 0;
+      else {
+        s = (size_t)read(fd, buf, sizeof(buf));
         close(fd);
       }
+      if (s == sizeof(buf))
+        s = *((size_t *) buf);
       else
 #endif /* USE_DEV_RANDOM */
         s = (size_t)(time(0) ^ (size_t)0x55555555U);
```